test/serializers/api/tracks_serializer.rb

```diff
@@ -0,0 +1,67 @@
+require 'test_helper'
+
+class API::TracksSerializerTest < ActiveSupport::TestCase
+  test "without user" do
+    track = create :track
+
+    num_concept_exercises = 3
+    num_practice_exercises = 7
+
+    num_concept_exercises.times { create :concept_exercise, track: track }
+    num_practice_exercises.times { create :practice_exercise, track: track }
+
+    serializer = API::TracksSerializer.new([track])
+    expected = {
+      tracks: [
+        {
+          id: track.id,
+          title: track.title,
+          num_concept_exercises: num_concept_exercises,
+          num_practice_exercises: num_practice_exercises,
+          web_url: "https://test.exercism.io/tracks/#{track.slug}",
+
+          # TODO: Set all three of these
+          icon_url: "https://assets.exercism.io/tracks/ruby-hex-white.png",
+          is_new: true,
+          tags: ["OOP", "Web Dev"]
+        }
+      ]
+    }
+
+    assert_equal expected, serializer.to_hash
+  end
+
+  test "with user not joined" do
```

Comment:
As I was reading this test, it's somewhat telling me that we could split this into two objects. One dealing with the context of not having a user, and one dealing with the context of having a user. It seems that this is testing too much as we're testing two cases within the user context.

We don't need to change it right now. It's just something that I noticed.

Reply:
Yeah, maybe. My reluctance to do that was that then we have to encode the same logic in multiple places to determine which serializer to call, and so the responsibility moves to multiple non-DRY controllers. Whereas I think what's actually changing is the output of the serialization based on various criteria (e.g. whether there is a user) - so it feels to me like the logic should be encapsualted within that serialization logic.

Reply:
Exactly why I think this isn't just a plain serializer.

Reply:
Yeah - I agree. We need a better name I think.